app/shop.py

```diff
@@ -0,0 +1,44 @@
+from __future__ import annotations
+
+import datetime
+from math import ceil
+
+
+class Shop:
+    def __init__(self,
+                 name: str,
+                 products: dict[str, float],
+                 location: list[int]) -> None:
+        self.name = name
+        self.products = products
+        self.location = location
+
+    @classmethod
+    def from_dict(cls, shop: dict) -> Shop:
+        return cls(name=shop["name"],
+                   products=shop["products"],
+                   location=shop["location"])
+
+    def is_in_stock(self, products: dict[str, float]) -> bool:
+        return all([product in self.products.keys()
+                    for product, count in products.items()])

```

Comment:
👍🏻

Reply:
By the way @ketstap162, you can remove `[]` here. It would do the same thing, but faster and more memory-efficient.

Reply:
yes, you are right 💯